src/bms/parse.rs

```diff
@@ -109,38 +116,123 @@ impl<'a> DerefMut for BmsParseTokenIter<'a> {
+            }
+        }
+
+        (warnings, errors)
+    }
+
     /// Parses a token stream into [`Bms`] with a random generator [`Rng`].
     pub fn from_token_stream<'a>(
         token_iter: impl Into<BmsParseTokenIter<'a>>,
         rng: impl Rng,
         mut prompt_handler: impl PromptHandler,
     ) -> BmsParseOutput {
-        let (continue_tokens, mut errors) = parse_control_flow(&mut token_iter.into(), rng);
+        let (continue_tokens, mut parse_warnings) = parse_control_flow(&mut token_iter.into(), rng);
         let mut notes = Notes::default();
         let mut header = Header::default();
         let mut non_command_lines: Vec<String> = Vec::new();
         let mut unknown_command_lines: Vec<String> = Vec::new();
         for &token in continue_tokens.iter() {
             if let Err(error) = notes.parse(token, &header) {
-                errors.push(error);
+                parse_warnings.push(error);
             }
             if let Err(error) = header.parse(token, &mut prompt_handler) {
-                errors.push(error);
+                parse_warnings.push(error);
             }
             match token {
                 Token::NotACommand(comment) => non_command_lines.push(comment.to_string()),
                 Token::UnknownCommand(comment) => unknown_command_lines.push(comment.to_string()),
                 _ => (),
             }
         }
+        let bms = Self {
+            header,
+            notes,
+            non_command_lines,
+            unknown_command_lines,
+        };
+
+        let (playing_warnings, playing_errors) = bms.check_playing_conditions();
+
         BmsParseOutput {
-            bms: Self {
-                header,
-                notes,
-                non_command_lines,
-                unknown_command_lines,
-            },
-            warnings: errors,
+            bms,
+            parse_warnings,
+            playing_warnings,
+            playing_errors,
         }
     }
 }
+
+/// Simpifies the warnings for playing, which would not make this chart unplayable.
+#[non_exhaustive]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Error)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum PlayingWarning {
+    /// The `#TOTAL` is not specified.
+    #[error("The `#TOTAL` is not specified.")]
+    TotalUndefined,
+    /// There is no displayable notes.
+    #[error("There is no displayable notes.")]
+    NoDisplayableNotes,
+    /// There is no playable notes.
+    #[error("There is no playable notes.")]
+    NoPlayableNotes,
+    /// The `#STARTBPM` is not specified. If there are other bpm changes, the first one will be used.
+    /// If there are no bpm changes, there will be an [`PlayingError::BpmUndefined`].
+    #[error(
+        "The `#STARTBPM` is not specified. If there are other bpm changes, the first one will be used. If there are no bpm changes, there will be an [`PlayingError::BpmUndefined`]."
+    )]
```

Comment:
```suggestion
    /// The `#BPM` is not specified. If there are other bpm changes, the first one will be used.
    /// If there are no bpm changes, there will be an [`PlayingError::BpmUndefined`].
    #[error(
        "The `#BPM` is not specified. If there are other bpm changes, the first one will be used. If there are no bpm changes, there will be an [`PlayingError::BpmUndefined`]."
    )]
```
There is no command `#STARTBPM`. It is `#BPM`, isn't it?